app/controllers/sessions_controller.rb

```diff
@@ -3,21 +3,32 @@ def new
   end
 
   def create
-    @user = User.find_by(email: params[:session][:email])
-    if @user && @user.admin? && @user.authenticate(params[:session][:password])
-      flash[:notice] = "Succsessfully Logged In As #{@user.username}"
-      session[:user_id] = @user.id
-      redirect_to admin_dashboard_path
-    elsif @user && @user.authenticate(params[:session][:password])
-      flash[:notice] = "Succsessfully Logged In As #{@user.username}"
-      session[:user_id] = @user.id
-      redirect_to user_path
+    if request.env['omniauth.auth']
+      @user = User.find_or_create_by_auth(request.env['omniauth.auth'])
+        flash[:notice] = "Succsessfully Logged In As #{@user.username}"
+        session[:user_id] = @user.id
+        redirect_to user_path
     else
-      flash[:error] = "Invalid C#ff0000entials. Please Try Again."
-      render :new
+      @user = User.find_by(email: params[:session][:email])
+      if @user && @user.admin? && @user.authenticate(params[:session][:password])
+        flash[:notice] = "Succsessfully Logged In As #{@user.username}"
+        session[:user_id] = @user.id
+        redirect_to admin_dashboard_path
+      elsif @user && @user.authenticate(params[:session][:password])
+        flash[:notice] = "Succsessfully Logged In As #{@user.username}"
+        session[:user_id] = @user.id
+        redirect_to user_path
+      else
+        flash[:error] = "Invalid Credentials. Please Try Again."
+        render :new
+      end
     end
   end
 
+  def twitter
+    redirect_to '/auth/twitter'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.